### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by EduardoContreras**
**Summary**

.- This Pull Request contains sign-up user on endpoint POST with all required validations of trello card and was merged with previous branches 'kickOff' and 'create-user-model'.

**Trello Card**
https://trello.com/c/GWo93nQs/2-sign-up

### CODE REVIEW

File: app/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const { map } = require('lodash'),
+  Model = require('../models/index'),
+  logger = require('../../app/logger');
+
+exports.updateUser = (req, res) => {
+  Model.sequelize.sync().then(function() {
```
Comment (gonziesc) [inline app/controllers/user.js:6]: no need of this here

File: app/controllers/user.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        return res.status(200).send({ message: 'User was updated', user });
+      })
+      .catch(err => {
+        logger.error('User was not updated. ', err.message);
+        return res.status(422).send({ message: map(err.errors, 'message') });
+      });
+  });
+};
+exports.deleteUser = (req, res) => {
+  Model.sequelize.sync().then(function() {
```
Comment (gonziesc) [inline app/controllers/user.js:24]: same

File: app/controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        return res.status(200).send(user);
+      })
+      .catch(err => {
+        logger.error('User was not found. ', err.message);
+        return res.status(422).send({ message: map(err.errors, 'message') });
+      });
+  });
+};
+exports.getUsers = (req, res) => {
+  Model.sequelize.sync().then(function() {
```
Comment (gonziesc) [inline app/controllers/user.js:52]: same

File: app/controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        return res.status(200).send(users);
+      })
+      .catch(err => {
+        logger.error('Lis of user was not found. ', err.message);
+        return res.status(422).send({ message: map(err.errors, 'message') });
+      });
+  });
+};
+exports.postUser = (req, res) => {
+  Model.sequelize.sync().then(function() {
```
Comment (gonziesc) [inline app/controllers/user.js:65]: same

File: app/controllers/user.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
+const { map } = require('lodash'),
+  Model = require('../models/index'),
```
Comment (gonziesc) [inline app/controllers/user.js:2]: you can import User

File: app/models/user.js [hunk 6]
```diff
@@ -42,9 +58,52 @@ module.exports = (sequelize, DataTypes) => {
       }
     },
     {
-      indexes: [{ unique: true, fields: ['id'] }, { unique: true, fields: ['email'] }]
+      indexes: [{ unique: true, fields: ['id'] }, { unique: true, fields: ['email'] }],
+      validate: {
+        isWoloxDomain() {
+          if (
```
Comment (gonziesc) [inline app/models/user.js:64]: validate this in the controller or a service, not in the database

File: documentation.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const dictum = require('dictum.js');
```
Comment (gonziesc) [inline documentation.js:1]: no need of this

File: test/factories/user.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+const faker = require('faker'),
```
Comment (gonziesc) [inline test/factories/user.js:1]: nice!

**Review by gonziesc — CHANGES_REQUESTED** [review]

File: app/models/user.js (hunk 6 above)
Comment (EduardoContreras) [reply app/models/user.js:64]: But, the approach is oriented to DDD ( domain driver desing)... all validation are over the model. Its not valid? And the last point before save to db is in the model.

**Pull request closed without merging**